grease-cli/tests/Shape.hs

```diff
@@ -85,6 +87,14 @@ eqShapes eqExt c1 c2 =
       eqShape eqExt x1 x2 && eqShapes eqExt c1' c2'
     (_, _) -> False
 
+eqMemShape :: PtrShape.MemShape w NoTag -> PtrShape.MemShape w NoTag -> Bool
+eqMemShape (PtrShape.Pointer ty off tgt) (PtrShape.Pointer ty' off' tgt') = ty == ty' && off == off' && eqPtrTarget tgt tgt'
+eqMemShape x y = x == y
+
+eqPtrTarget :: PtrShape.PtrTarget w NoTag -> PtrShape.PtrTarget w NoTag  -> Bool
+eqPtrTarget (PtrShape.PtrTarget mems _)  (PtrShape.PtrTarget mems' _) =
+  Seq.length mems == Seq.length mems' && foldl (\acc (x,y) -> acc && eqMemShape x y) True (Seq.zip mems mems')
```

Comment:
Worth a comment about what is being ignored and why. Also, long line! Also, consider `all`:
```suggestion
eqPtrTarget :: PtrShape.PtrTarget w NoTag -> PtrShape.PtrTarget w NoTag  -> Bool
eqPtrTarget (PtrShape.PtrTarget mems _)  (PtrShape.PtrTarget mems' _) =
  Seq.length mems == Seq.length mems' &&
    all (uncurry eqMemShape) (Seq.zip mems mems')
```